server/controllers/articles.js

```diff
@@ -0,0 +1,103 @@
+import { validationResponse } from '@helpers/validationResponse';
+import Response from '@helpers/Response';
+import models from '@models';
+import validateRating from '@validations/rating';
+
+const { User, Article, Rating } = models;
+
+/**
+ * @exports ArticleController
+ * @class ArticleController
+ * @description Handles Article related actions
+ * */
+class ArticleController {
+  /**
+   * Rate an article
+   * @async
+   * @param  {object} req - Request object
+   * @param {object} res - Response object
+   * @param {object} next The next middleware
+   * @return {json} Returns json object
+   * @static
+   */
+  static async rate(req, res, next) {
+    const userId = req.decoded.id;
+
+    try {
+      // Validate the rating
+      const articleDetails = await validateRating(req.body);
+
+      const { articleId, rate } = articleDetails;
+
+      // Check if the article exists
+      const article = await Article.findOne({
+        where: { id: articleId }
+      });
+
+      if (!article) return Response.error(res, 404, 'Article does not exist');
+
+      // Get the full details of user and article
+      const moreDetails = await Article.findOne({
+        where: { id: articleId },
+        attributes: { exclude: ['updatedAt', 'createdAt'] },
+        include: [
+          {
+            model: User,
+            as: 'author',
+            attributes: { exclude: ['password', 'updatedAt', 'createdAt'] }
```

Comment:
please include only the fields you need here rather than excluding the ones you do not need